ethereum/spec/src/testFixtures/java/tech/pegasys/teku/spec/datastructures/util/MerkleTree.java

```diff
@@ -113,25 +114,25 @@ public List<Bytes32> getProof(int itemIndex) {
         proof.add(zeroHashes.get(i));
       }
 
-      itemIndex /= 2;
+      mutableIndex /= 2;
     }
     proof.add(calcMixInValue());
     return proof;
   }
 
-  private Bytes32 calcViewBoundaryRoot(int depth, int viewLimit) {
+  private Bytes32 calcViewBoundaryRoot(final int depth, final int viewLimit) {
     if (depth == 0) {
       return zeroHashes.get(0);
     }
-    depth -= 1;
-    Bytes32 deeperRoot = calcViewBoundaryRoot(depth, viewLimit);
+    int mutableDepth = depth - 1;
```

Comment:
wait so this is just off by 1? looks like this is not altered after decrementing.

Reply:
you're right this is just single decrement operation done on top of the input param `depth`. New variable can actually be `final` as well

Reply:
probably worth just looking where this is called too and see whether the depth passed in is just wrong lol

Reply:
So mutation of this index is actually part of the recursive approach used in the this method.

Reply:
i do still think final, and potentially deeperRoot could be final also

so mutableDepth => final internalDepth
and deeperRoot -> final... at least on the code here...